extensions/jdbc/jdbc-postgresql/deployment/src/main/java/io/quarkus/jdbc/postgresql/deployment/PostgreSQLJDBCReflections.java

```diff
@@ -19,6 +19,9 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         //We register it for the sake of other users.
         final String driverName = "org.postgresql.Driver";
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, driverName));
+
+        // Needed when quarkus.datasource.jdbc.transactions=xa for the setting of the username and password
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, true, false, "org.postgresql.ds.common.BaseDataSource"));
```

Comment:
I wonder if we should have a better infrastructure to push the classes to the JDBC build item and get the JDBC infrastructure do the work. But maybe let's wait to have another similar case before doing it.